sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/reflect/DoFnSignaturesTest.java

```diff
@@ -263,4 +392,15 @@ public void foo(ProcessContext context) {}
     @ProcessElement
     public void foo(ProcessContext context) {}
   }
+
+  private static class DoFnForTestSimpleTimerIdNamedDoFn extends DoFn<KV<String, Integer>, Long> {
+    @TimerId("foo")
+    private final TimerSpec bizzle = TimerSpecs.timer(TimeDomain.EVENT_TIME);
+
+    @ProcessElement
+    public void foo(ProcessContext context) {}
+
+    @OnTimer("foo")
+    public void onFoo() {}
+  }
```

Comment:
My stylistic preference would be to disallow this. I think the declaration and use should be in the same lexical scope. But I suppose users want what they want...